opm/output/eclipse/summaryComparator.cpp

```diff
@@ -0,0 +1,243 @@
+ 
+  for (int time_index = 0; time_index < ecl_sum_get_data_length(ecl_sum1); time_index++){
+    timeVec1.push_back(ecl_sum_iget_sim_days(ecl_sum1 , time_index ));
+  }
+  
+
+  for (int time_index = 0; time_index < ecl_sum_get_data_length(ecl_sum2); time_index++){
+    timeVec2.push_back(ecl_sum_iget_sim_days(ecl_sum2 , time_index ));
+  }
+}
+
+
+
+
+//Read the data from the two files into separate vectors. Not necessary the same amount of values, but the values correspond to the same interval in time. Thus possible to interpolate values.
+ 
+void SummaryComparator::getDataVecs(std::vector<double> &dataVec1,std::vector<double> &dataVec2,const char* keyword){  
+  for (int time_index = 0; time_index < ecl_sum_get_data_length(ecl_sum1); time_index++){
+    dataVec1.push_back(ecl_sum_iget(ecl_sum1, time_index, ecl_sum_get_general_var_params_index( ecl_sum1 , keyword )));
+  }
+  for (int time_index = 0; time_index < ecl_sum_get_data_length(ecl_sum2); time_index++){
+    dataVec2.push_back(ecl_sum_iget(ecl_sum2, time_index, ecl_sum_get_general_var_params_index( ecl_sum2 , keyword )));
+  }
+   
+}
+
+
+void SummaryComparator::setDataSets(std::vector<double> &timeVec1,std::vector<double> &timeVec2){
+  if(timeVec1.size()< timeVec2.size()){
+    ecl_sum_fileShort = this->ecl_sum1;
+    ecl_sum_fileLong = this->ecl_sum2;
+  }
+  else{
+    ecl_sum_fileShort = this->ecl_sum2;
+    ecl_sum_fileLong = this->ecl_sum1;
+  }
+}
+   
+
+
+//Figures out which time vector that contains the fewer elements. Sets this as referenceVec and its corresponding 
+// data as referenceDataVec. The other vector-set as checkVec( the time vector) and checkDataVec.
+void SummaryComparator::chooseReference(std::vector<double> &timeVec1,std::vector<double> &timeVec2,std::vector<double> &dataVec1,std::vector<double> &dataVec2){
+  if(timeVec1.size() <= timeVec2.size()){
+    referenceVec = &timeVec1; // time vector
+    referenceDataVec = &dataVec1; //data vector
+    checkVec = &timeVec2;
+    checkDataVec = &dataVec2;     
+  }
+  else{
+    referenceVec = &timeVec2;
+    referenceDataVec = &dataVec2;
+    checkVec = &timeVec1;
+    checkDataVec = &dataVec1;
+  }
+
+
+}
+
+
+//Calculates each deviation and stores it in a deviation struct. Uses the referenceVec as basis, and checks its values against the values int checkDataVec
+//Should be called inside a for/while loop running over the reference vector, with ivar as the iterative variable of the loop. 
+// Updates iteration variable jvar aswell, which will be used for the checkVec/cecheDataVec, thus will not check same keyword twice
+void SummaryComparator::getDeviation(int refIndex, int &checkIndex, Deviation &dev){
+  if((*referenceVec)[refIndex] == (*checkVec)[checkIndex]){
+    dev = SummaryComparator::calculateDeviations((*referenceDataVec)[refIndex], (*checkDataVec)[checkIndex]); 
+    checkIndex++;
+    return;
+  }
+  else if((*referenceVec)[refIndex]<(*checkVec)[checkIndex]){
```

Comment:
Of course not  :-) I still suggest we go with a simple step function first and see how that works out.  We may not need more sophistication. Since pressure and rate curves has discontinuities at well changes, a simple linear interpolation will not work.